ddtrace/contrib/grpc/interceptors/_interceptor.py

```diff
@@ -0,0 +1,343 @@
+
+
+class _InterceptingChannel(grpc.Channel):
+
+    def __init__(self, channel, interceptor):
+        self._channel = channel
+        self._interceptor = interceptor
+
+    def subscribe(self, *args, **kwargs):
+        self._channel.subscribe(*args, **kwargs)
+
+    def unsubscribe(self, *args, **kwargs):
+        self._channel.unsubscribe(*args, **kwargs)
+
+    def unary_unary(self,
+                    method,
+                    request_serializer=None,
+                    response_deserializer=None):
+
+        def callable_factory(method):
+            return self._channel.unary_unary(method, request_serializer,
+                                             response_deserializer)
+
+        if isinstance(self._interceptor, grpc.UnaryUnaryClientInterceptor):
+            return _InterceptingUnaryUnaryMultiCallable(
+                method, callable_factory, self._interceptor)
+        else:
+            return callable_factory(method)
+
+    def unary_stream(self,
+                     method,
+                     request_serializer=None,
+                     response_deserializer=None):
+
+        def callable_factory(method):
+            return self._channel.unary_stream(method, request_serializer,
+                                              response_deserializer)
+
+        if isinstance(self._interceptor, grpc.UnaryStreamClientInterceptor):
+            return _InterceptingUnaryStreamMultiCallable(
+                method, callable_factory, self._interceptor)
+        else:
+            return callable_factory(method)
+
+    def stream_unary(self,
+                     method,
+                     request_serializer=None,
+                     response_deserializer=None):
+
+        def callable_factory(method):
+            return self._channel.stream_unary(method, request_serializer,
+                                              response_deserializer)
+
+        if isinstance(self._interceptor, grpc.StreamUnaryClientInterceptor):
+            return _InterceptingStreamUnaryMultiCallable(
+                method, callable_factory, self._interceptor)
+        else:
+            return callable_factory(method)
+
+    def stream_stream(self,
+                      method,
+                      request_serializer=None,
+                      response_deserializer=None):
+
+        def callable_factory(method):
+            return self._channel.stream_stream(method, request_serializer,
+                                               response_deserializer)
+
+        if isinstance(self._interceptor, grpc.StreamStreamClientInterceptor):
+            return _InterceptingStreamStreamMultiCallable(
+                method, callable_factory, self._interceptor)
+        else:
+            return callable_factory(method)
+
+
+def intercept_channel(channel, *interceptors):
+    for interceptor in reversed(list(interceptors)):
+        if not isinstance(interceptor, grpc.UnaryUnaryClientInterceptor) and \
```

Comment:
You can use `isinstance(interceptor, (grpc.UnaryUnaryClientInterceptor, grpc.UnaryStreamClientInterceptor, grpc.StreamUnaryClientInterceptor, grpc.StreamStreamClientInterceptor))`

It might also be simpler to define a `ClientInterceptor` base class just to simplify this check. Would help just in case we need to add more in the future also.